src/EditorFeatures/Test2/Expansion/NameExpansionTests.vb

```diff
@@ -148,6 +148,119 @@ namespace NS
             Test(input, expected)
         End Sub
 
+        <WpfFact, Trait(Traits.Feature, Traits.Features.Expansion)>
+        Public Sub CSharp_GenericNameExpansion_DontExpandAnonymousTypes()
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+class C
+{
+    static void Mumble&lt;T&gt;(T anonymousType) { }
+
+    static void M()
+    {
+        {|Expand:Mumble|}(new { x = 42 });
+    }
+}
+        </Document>
+    </Project>
+</Workspace>
+
+            Dim expected =
+<code>
+class C
+{
+    static void Mumble&lt;T&gt;(T anonymousType) { }
+
+    static void M()
+    {
+        global::C.Mumble(new { x = 42 });
+    }
+}
+</code>
+
+            Test(input, expected)
+        End Sub
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.Expansion)>
+        Public Sub CSharp_LambdaParameter_DontExpandAnonymousTypes1()
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+using System;
+class C
+{
+    static void Mumble&lt;T&gt;(T anonymousType, Action&lt;T, int&gt; lambda) { }
+
+    static void M()
+    {
+        Mumble(new { x = 42 }, {|Expand:a => a.x|});
+    }
+}
+        </Document>
+    </Project>
+</Workspace>
+
+            Dim expected =
+<code>
+using System;
+class C
+{
+    static void Mumble&lt;T&gt;(T anonymousType, Action&lt;T, int&gt; lambda) { }
+
+    static void M()
+    {
+        Mumble(new { x = 42 }, a => a.x);
+    }
+}
+</code>
+
+            Test(input, expected, expandParameter:=True)
+        End Sub
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.Expansion)>
+        Public Sub CSharp_LambdaParameter_DontExpandAnonymousTypes2()
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+using System;
+class C
+{
+    static void Mumble&lt;T&gt;(T anonymousType, Action&lt;T, int, int&gt; lambda) { }
+
+    static void M()
+    {
+        Mumble(new { x = 42 }, {|Expand:(a, y) => a.x|});
```

Comment:
Is the lambda expecting three parameters here to compile? It's a Action<T, int, int>, not Func<T, int, int>